src/lualib/Error.ts

```diff
@@ -0,0 +1,82 @@
+type TSTLCapturedErrorStack = Array<{
+    namewhat: string;
+    name?: string;
+    source: string;
+    short_src: string;
+    currentline: number;
+}>;
+
+function __TS__GetErrorStack(constructor: Function): TSTLCapturedErrorStack {
+    const functionFrames = [];
+    let level = 1;
+    while (true) {
+        const info = debug.getinfo(level, "f");
+        level += 1;
+        if (!info) {
+            // constructor not in call stack
+            level = 1;
+            break;
+        } else if (info.func === constructor) {
+            break;
+        }
+    }
+    while (true) {
+        const info = debug.getinfo(level, "Snl");
+        if (!info) break;
+        if (info.currentline !== -1) {
+            functionFrames[functionFrames.length] = info;
+        }
+        level += 1;
+    }
+    return functionFrames;
+}
+
+function __TS__ConvertErrorStack(stack: TSTLCapturedErrorStack): string {
+    const info = stack
+        .map(v => {
+            if (v.namewhat === "") {
+                return `${v.short_src}:${v.currentline}`;
+            } else {
+                return `${v.short_src}:${v.currentline} in ${v.namewhat} ${v.name}`;
+            }
+        })
+        .join("\n");
+    const transform = globalThis.__TS__SourceMapTransform;
```

Comment:
Would this not require a sourcemap traceback dependency?

Reply:
Not really. `globalThis.__TS__SourceMapTransform` is defined by default if and only if source maps are enabled. If source maps are enabled the function exists and we transform the traceback. Otherwise we keep the original stack unmodified.

Reply:
Not true, you defined __TS__SourceMapTransform in the SourceMapTraceBack lualib entry, which is only defined if sourceMapTraceBack is enabled in options (regardless of source maps). That said, I think the way you are handling this is correct since you only want to apply it at the moment sourceMapTraceBack is defined. That said, I would like to see a test for this with both sourceMapTraceBack enabled and disabled.

Reply:
Ah, so _sourceMapTraceBack_ is what I meant to say then :D